android/envoy/src/main/kotlin/org/greatfire/envoy/NetworkIntentService.kt

```diff
@@ -483,34 +469,34 @@ class NetworkIntentService : IntentService("NetworkIntentService") {
 
     private fun handleHttpsSubmit(
         url: String,
-        captive_portal_url: String,
-        hysteriaCert: String?
+        captive_portal_url: String
     ) {
-
-        // nothing to parse at this time, leave url in string format
-
-        httpsUrls.add(url)
         Log.d(TAG, "submit http(s) url")
-        handleRequest(url, url, ENVOY_SERVICE_HTTPS, captive_portal_url, hysteriaCert)
-    }
 
-    private fun handleEnvoySubmit(
-        url: String,
-        captive_portal_url: String,
-        hysteriaCert: String?
-    ) {
+        val p = Plenipotentiary.newPlenipotentiary()
+        p.envoyUrl = url
+        var newUrl = p.findEnvoyUrl(filesDir.path)
+        httpsUrls.add(newUrl)
 
-        // nothing to parse at this time, leave url in string format
+        Log.d(TAG, "using Envoy URL: " + newUrl)
 
-        httpsUrls.add(url)
-        Log.d(TAG, "submit envoy url")
-        handleRequest(url, url, ENVOY_SERVICE_ENVOY, captive_portal_url, hysteriaCert)
+        handleRequest(url, newUrl, ENVOY_SERVICE_HTTPS, captive_portal_url)
     }
 
+    // private fun handleEnvoySubmit(
+    //     url: String,
+    //     captive_portal_url: String
+    // ) {
+
+    //     // nothing to parse at this time, leave url in string format
+    //     httpsUrls.add(url)
+    //     Log.d(TAG, "submit envoy url")
+    //     handleRequest(url, url, ENVOY_SERVICE_ENVOY, captive_portal_url)
+    // }
+
     private fun handleShadowsocksSubmit(
```

Comment:
could remove this along with all the rest, it doesn't look like it's called anymore.